i18n-helpers/fuzz/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+use arbitrary::Arbitrary;
+use mdbook::book::{Book, BookItem, Chapter};
+use polib::catalog::Catalog;
+use polib::message::Message;
+use polib::metadata::CatalogMetadata;
+use std::path::PathBuf;
+
+/// Generate a random Catalog for fuzzing.
+pub fn create_catalog(translations: Vec<(&str, &str)>) -> Catalog {
+    let mut catalog = Catalog::new(CatalogMetadata::new());
+    for (idx, (msgid, msgstr)) in translations.iter().enumerate() {
+        let message = Message::build_singular()
+            .with_source(format!("foo.md:{idx}"))
+            .with_msgid(String::from(*msgid))
+            .with_msgstr(String::from(*msgstr))
+            .done();
+        catalog.append_or_update(message);
+    }
+    catalog
+}
+
+/// Generate a random Book for fuzzing.
+pub fn create_book(book_items: Vec<MyBookItem>) -> Book {
+    let mut book = Book::new();
+    book_items.into_iter().for_each(|item| {
+        book.push_item(item);
+    });
+    book
+}
+
+/// Wrapper enum for generating arbitrary `BookItem`s.
+#[derive(Arbitrary, Debug)]
+pub enum MyBookItem {
```

Comment:
Please call it `BookItem`, I don't use the `My` prefix much outside of personal toy projects.

Then the `From` impl below will end up looking like

```rust
impl From<BookItem> for mdbook::book::BookItem
```

which nicely shows how the local type (unqualified) relates to the foreign type.

Reply:
Done.